store/src/java/com/zimbra/cs/imap/ImapPath.java

```diff
@@ -560,6 +560,10 @@ boolean isSelectable() throws ServiceException {
 
     boolean isVisible() throws ServiceException {
         boolean isMailFolders = Provisioning.getInstance().getLocalServer().isImapDisplayMailFoldersOnly();
+        if(folder == null) {
+            ZimbraLog.imap.error("ImapFolderStore is null. This session was likely terminated.");
```

Comment:
This scenario will happen when remote server restarts and it causes IMAP client session to be closed. So far I only tested this with Apple Mail and Apple Mail did not complain, it quietly created a new session. I suspect some IMAP clients may complain, so I would leave this as error until we get more data from testing in our hosting environment.